packages/api/endpoints/pdrs.js

```diff
@@ -63,28 +63,46 @@ const isRecordDoesNotExistError = (e) => e.message.includes('RecordDoesNotExist'
  * @returns {Promise<Object>} the promise of express response object
  */
 async function del(req, res) {
-  const pdrName = req.params.pdrName;
+  const {
+    pdrModel = new models.Pdr(),
+    pdrPgModel = new PdrPgModel(),
+    knex = await getKnexClient(),
+    esClient = await Search.es(),
+  } = req.testContext || {};
 
+  const pdrName = req.params.pdrName;
   const pdrS3Key = `${process.env.stackName}/pdrs/${pdrName}`;
 
-  const pdrModel = new models.Pdr();
-  const pdrPgModel = new PdrPgModel();
-  const knex = await getKnexClient();
-
+  let existingPdr;
   try {
-    await knex.transaction(async (trx) => {
-      await pdrPgModel.delete(trx, { name: pdrName });
-      await deleteS3Object(process.env.system_bucket, pdrS3Key);
-      await pdrModel.delete({ pdrName });
-    });
+    existingPdr = await pdrModel.get({ pdrName });
+  } catch (error) {
+    // Ignore error if record does not exist in DynamoDb
+    if (!(error instanceof RecordDoesNotExist)) {
+      throw error;
+    }
+  }
 
-    if (inTestMode()) {
-      const esClient = await Search.es(process.env.ES_HOST);
-      await esClient.delete({
-        id: pdrName,
-        index: process.env.ES_INDEX,
-        type: 'pdr',
-      }, { ignore: [404] });
+  try {
+    try {
+      await knex.transaction(async (trx) => {
+        await pdrPgModel.delete(trx, { name: pdrName });
+        await deleteS3Object(process.env.system_bucket, pdrS3Key);
+        await pdrModel.delete({ pdrName });
+        await esClient.delete({
+          id: pdrName,
+          index: process.env.ES_INDEX,
+          type: 'pdr',
+          refresh: inTestMode(),
+        }, { ignore: [404] });
+      });
+    } catch (innerError) {
```

Comment:
Do we need to do anything here with `deleteS3Object`? Or is it understood that if there's an error we will have already deleted the S3 file?

Reply:
Yeah, I'm not sure what we can do about it. But really this PR is only concerned with the affects of adding the ES write into the transaction and handling any side effects of that. 

But once an object is deleted from s3, you can't recreate it without having the contents of the original file. We could perhaps add something to handle it, but it gets pretty complicated with 4 operations in the transaction.

Reply:
So I took a quick look at adding something to recreate the S3 file if any of the delete operations fail, but it's the same thing where isolating whether it was the S3 delete that failed or something else would be pretty hard, so we'd have to just do a blanket PUT of the S3 file if any of the delete operations failed. Thoughts?

Reply:
Yeah, hmm recreating the S3 files seems less than desirable. Could `deleteS3Object` be the last operation in the transaction? That way we've definitely removed the PDR from PG, dynamo, and ES before deleting from S3. It's a tradeoff because then, if S3 delete fails we have a file in S3 that doesn't exist anywhere else. I suppose it's a question of which is preferable as a worst-case scenario.

Reply:
How about this?

https://github.com/nasa/cumulus/pull/2234/commits/47973f35030a7f4f3e6c6254c1e6fbbe1a9b6613

Reply:
Nice. Resolved with : https://github.com/nasa/cumulus/pull/2234#commitcomment-51897225